### PR TITLE
Multi replica

### PR DESCRIPTION
**Pull request by ozgunozerk**
Creating as a draft PR at the moment. 
When we do the runtime upgrade on the monorepo side, I will remove the custom chainspec file from here, and turn it into a real PR

### CODE REVIEW

File: src/loc/en.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "utilizedSpace": "How much space is currently being used.",
     "allocatedSpace": "Space to be shared with Subspace Protocol.",
     "estimatingSpace": "Requesting the blockchain size from the network.",
+    "nodeSpace": "Space Requied for Node",
```
Comment (ozgunozerk) [inline src/loc/en.json:33]: note to myself: also get rid of this when deleting the custom chain-spec file

File: src-tauri/src/farmer.rs [hunk 2]
```diff
@@ -31,52 +17,27 @@ pub(crate) async fn farm(
...
 
     info!("Connecting to node at {}", node_rpc_url);
-    let client = WsRpc::new(node_rpc_url).await?;
-
-    let farmer_metadata = client
+    let client = WsRpc::new(&node_rpc_url).await?;
+
+    let FarmerMetadata {
+        record_size: _,
+        recorded_history_segment_size: _,
```
Comment (i1i1) [inline src-tauri/src/farmer.rs:35]: Should those 2 lines be removed?

File: src-tauri/src/main.rs [hunk 3]
```diff
@@ -36,15 +39,27 @@ fn plot_progress_tracker() -> usize {
 async fn farming(path: String, reward_address: String, plot_size: u64) {
```
Comment (i1i1) [inline src-tauri/src/main.rs]: Btw, does tauri support `Option<String>` and `PathBuf` for arguments? It would be handy to use those here

Comment (ozgunozerk) [reply src-tauri/src/main.rs]: unfortunately no :/ 
All the parameters and return types for `tauri command` functions need to be primitive types, since these are communicating with frontend.

Comment (i1i1) [reply src-tauri/src/main.rs]: I believe `Option<String>` at least should be? There is `null` in js

File: src-tauri/src/farmer.rs (hunk 2 above)
Comment (ozgunozerk) [reply src-tauri/src/farmer.rs:35]: agreed, thanks!

File: src-tauri/src/main.rs (hunk 3 above)
Comment (ozgunozerk) [reply src-tauri/src/main.rs]: Maybe, but I think it serves no purpose for desktop app. Since there **always** will be `reward_address` and `path`. 

The reason for mandatory reward address: 
the first case: if the user decides to import reward address -> user enters the reward address, and frontend sends the reward address to backend.
the second case: user does not import reward address -> because of multi-replica, there has to be a ROOT-IDENTITY for reward address. We choose to generate this new identity on the frontend, for unified workflow (frontend sending the reward address to backend). 

In summary, there is no path in our workflow to *NOT* send a reward address to backend.

Comment (i1i1) [reply src-tauri/src/main.rs]: Ok, should we delete branch there we check for `length == 0` on reward address?

Comment (i1i1) [reply src-tauri/src/main.rs]: Missed changes in latter commits. It is no longer needed.

**Review by i1i1 — APPROVED** [review]

**Comment by ozgunozerk** [comment]
Please give it a final look, it is ready for a real review now :)

File: src-tauri/src/node.rs [hunk 4]
```diff
@@ -148,63 +173,66 @@ fn create_configuration<CS: ChainSpec + 'static>(
...
-        tracing_targets: None,
-        tracing_receiver: TracingReceiver::Log,
-        chain_spec: Box::new(chain_spec),
-        max_runtime_instances: 8,
-        announce_block: true,
-        role,
-        base_path: Some(base_path),
-        informant_output_format: Default::default(),
-        runtime_cache_size: 2,
+        force_new_slot_notifications: false,
```
Comment (ozgunozerk) [inline src-tauri/src/node.rs:235]: Nazar, I don't not 100% sure this should be false/true. Can you double check?

File: src/lib/client.ts [hunk 5]
```diff
@@ -53,26 +54,22 @@ export class Client {
       },
       start: async (): Promise<void> => {
 
-        const farmerAddress = LocalStorage.getItem("rewardAddress")
-        if (farmerAddress === null) {
-          console.error("Reward address should not have been null...")
+        const farmerAddress: string | undefined = appConfig.getAppConfig()?.rewardAddress
+        if (farmerAddress == null) {
```
Comment (isSerge) [inline src/lib/client.ts:58]: `===` is usually preferred over `==`
in this case, you can just do `if (!farmerAddress) {`

File: src/pages/SetupPlot.vue [hunk 6]
```diff
@@ -299,10 +295,16 @@ export default defineComponent({
     },
     async checkIdentity() {
       const config = appConfig.getAppConfig()
-      if (config && config.importedRewAddr === false) {
-        await this.client.createRewardAddress()
-        await this.viewMnemonic()
+      if (config) {
+        if (config.rewardAddress == "") {
```
Comment (isSerge) [inline src/pages/SetupPlot.vue:299]: `if (!config.rewardAddress)` is better

File: src-tauri/src/node.rs (hunk 4 above)
Comment (ozgunozerk) [reply src-tauri/src/node.rs:235]: forgot to tag properly in the previous comment -> @nazar-pc

Comment (nazar-pc) [reply src-tauri/src/node.rs:235]: `false` is fine

File: src-tauri/src/node.rs [hunk 7]
```diff
@@ -148,63 +173,75 @@ fn create_configuration<CS: ChainSpec + 'static>(
...
+                other: ExecutionStrategy::AlwaysWasm,
+            },
+            #[cfg(target_os = "windows")]
+            execution_strategies: ExecutionStrategies {
+                syncing: ExecutionStrategy::NativeElseWasm,
+                importing: ExecutionStrategy::NativeElseWasm,
+                block_construction: ExecutionStrategy::NativeElseWasm,
+                offchain_worker: ExecutionStrategy::NativeElseWasm,
+                other: ExecutionStrategy::NativeElseWasm,
+            },
```
Comment (ozgunozerk) [inline src-tauri/src/node.rs:208]: these are the latest changes from release preparations (monorepo) @nazar-pc

File: src-tauri/src/node.rs [hunk 8]
```diff
@@ -148,63 +173,75 @@ fn create_configuration<CS: ChainSpec + 'static>(
...
+            network,
+            keystore_remote,
+            keystore,
+            database: database_config(&config_dir, database_cache_size, &role),
+            state_cache_size: 67_108_864,
+            state_cache_child_ratio: None,
+            // TODO: Change to constrained eventually (need DSN for this)
+            state_pruning: PruningMode::keep_blocks(1024),
+            keep_blocks: KeepBlocks::Some(1024),
+            wasm_method: WasmExecutionMethod::Interpreted,
```
Comment (nazar-pc) [inline src-tauri/src/node.rs:191]: Ouch, no interpreted WASM!!!

File: src/pages/SetupPlot.vue [hunk 9]
```diff
@@ -173,7 +172,7 @@ export default defineComponent({
     },
     stats(): StatsType {
       const totalDiskSizeGB = util.toFixed(this.driveStats.totalBytes / 1e9, 2)
-      const safeAvailableGB = this.driveStats.freeBytes / 1e9
+      const safeAvailableGB = this.driveStats.freeBytes / 1e9 - 20
```
Comment (nazar-pc) [inline src/pages/SetupPlot.vue:175]: What is this 20? We need to allocate 10GB for node and the rest for farmer, but the user should only specify the combined number, they only care how much will software use in total.

Comment (ozgunozerk) [reply src/pages/SetupPlot.vue:175]: We have talked with you previously that node is at most occupy 20GB. 
So I put 20GB for a safe cap. 

Jeremiah and I discussed that we should not complicate the user workflow and maybe should not even mention about that (node occupying space). So user can enter the plot size, and we can deduct 20GB from the safe space.

I can update this to 10 if you like, no problem. 
However, when we switch to full-node, will it still occupy around 10GB? I think it would be around 1-2GB at most? Am I recalling it wrong?

File: src-tauri/src/node.rs (hunk 8 above)
Comment (ozgunozerk) [reply src-tauri/src/node.rs:191]: Since we removed `wasm exeuction compiled` from readme
https://github.com/subspace/subspace/pull/405/files#diff-cb44c4563e0dbee8cb7ed4e05a25e2708bd167ef5b59cabc8e9fcad2dbb70e73L39

I thought it would use the default.

and the default for `WasmExecutionMethod` was ->
```rust
impl Default for WasmExecutionMethod {
	fn default() -> WasmExecutionMethod {
		WasmExecutionMethod::Interpreted
	}
}
```

So we are using `compiled` and it is declared in somewhere else?

**Review by nazar-pc — APPROVED** [review]

**Review by nazar-pc — APPROVED** [review]
